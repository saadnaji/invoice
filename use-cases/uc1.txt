add_type("nuts")
add_type("bolts")
add_type("HAMMERS")     
add_product("bolts", 100)
add_product("hammers", 12) 
add_product("nuts", 2000)
add_product("hammers", 7)                  
add_order(<<["bolts", 5], ["hammers", 1]>>)
cancel_order(1)   
add_order(<<["hammers", 1], ["bolts", 1], ["nuts", 9]>>)
add_order(<<["hammers", 2], ["nuts", 10]>>)
invoice(1)
