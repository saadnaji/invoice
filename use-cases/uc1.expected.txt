  report:      ok
  id:          0
  products:   
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("nuts")
  report:      ok
  id:          0
  products:    nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("bolts")
  report:      ok
  id:          0
  products:    bolts,nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_type("HAMMERS")
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       
  orders:      
  carts:       
  order_state: 
->add_product("bolts",100)
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100
  orders:      
  carts:       
  order_state: 
->add_product("hammers",12)
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->12
  orders:      
  carts:       
  order_state: 
->add_product("nuts",2000)
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->12,nuts->2000
  orders:      
  carts:       
  order_state: 
->add_product("hammers",7)
  report:      ok
  id:          0
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->19,nuts->2000
  orders:      
  carts:       
  order_state: 
->add_order(<<["bolts", 5], ["hammers", 1]>>)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->95,hammers->18,nuts->2000
  orders:      1
  carts:       1: bolts->5,hammers->1
  order_state: 1->pending
->cancel_order(1)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->100,hammers->19,nuts->2000
  orders:      
  carts:       
  order_state: 
->add_order(<<["hammers", 1], ["bolts", 1], ["nuts", 9]>>)
  report:      ok
  id:          1
  products:    bolts,hammers,nuts
  stock:       bolts->99,hammers->18,nuts->1991
  orders:      1
  carts:       1: bolts->1,hammers->1,nuts->9
  order_state: 1->pending
->add_order(<<["hammers", 2], ["nuts", 10]>>)
  report:      ok
  id:          2
  products:    bolts,hammers,nuts
  stock:       bolts->99,hammers->16,nuts->1981
  orders:      1,2
  carts:       1: bolts->1,hammers->1,nuts->9
               2: hammers->2,nuts->10
  order_state: 1->pending,2->pending
->invoice(1)
  report:      ok
  id:          2
  products:    bolts,hammers,nuts
  stock:       bolts->99,hammers->16,nuts->1981
  orders:      1,2
  carts:       1: bolts->1,hammers->1,nuts->9
               2: hammers->2,nuts->10
  order_state: 1->invoiced,2->pending
